src/Core/Silk.NET.BuildTools/Generator.cs

```diff
@@ -217,9 +217,16 @@ private static void RunTask(BindTask task, Config config, Stopwatch? sw)
                         }
                     }
                     ClangConfig.SubstituteWindowsSdkPath(ref task);
+                    Profile coreProfile = null;
                     foreach (var src in task.Sources)
                     {
-                        profiles.Add(Clang.GenerateProfile(Path.GetFileName(src), File.OpenRead(GetPath(src)), task));
+                        var createdProfile = Clang.GenerateProfile(Path.GetFileName(src), File.OpenRead(GetPath(src)), task, coreProfile);
+
+                        //If the created profile contains a root project, then its likely the core profile
+                        if(createdProfile.Projects.Any(x => x.Value.IsRoot))
```

Comment:
All profiles must contain a root project for baking reasons so this is redundant. iirc there is code in Clang to make this the case.